### DISCUSSION TITLE
Module federation security concerns

### DISCUSSION THREAD

**Discussion by ktsangop** [webpack/webpack#16230] [General]
Hello everyone!
I am reading about Module federation, and I would like to utilize it in one of our company's projects.

I am trying to find any security concerns that apply to Module Federation in general, and how to mitigate them.
* For example what are the security implications of replacing an `<iframe>` which embeds a microfrontend with a Federated module which does the same?
* Do federated modules adhere to the same-origin policy?
* Are there any specific best practices regarding security when deploying federated modules? 
* How would someone handle authentication/authorization in a federated module (remote) that is required to communicate with an API (let's suppose that the shell is already authorized to use the API before the remote is loaded)

Thanks in advance!

**Comment by alexander-akait** [comment]
@ScriptedAlchemy friendly ping

**Comment by ScriptedAlchemy** [comment]
If the shell is already authorized, id put that into a piece of state, like recoil, redux, react context - which is shared as a singleton so all other apps can see its authorized. 

The browser will respect origin policies - control security via CSP headers or whatever else. 
Security wise - not much different than any other application. Webpack uses chunk loading globals like window.webpackChunkLoad - so if someone hacks your site, they can take over the chunk loading global of webpack. Remote entry should pose no additional threats. If someone is already able to hijack anything in your application - you've already been hacked and federation is least of your concerns. 

Security best practices - same as you'd do to secure rogue JS running on your website. CSP policies so the browser is only allowed to accept resources from trusted domains.

**Reply by ktsangop** [reply]
Thank you @ScriptedAlchemy 
I just wanted to be sure that there are no extra security concerns.

The information you provide could be part of the official document page :
https://webpack.js.org/concepts/module-federation/

:)

**Reply by ScriptedAlchemy** [reply]
Send a PR :) 
I've got no capacity

**Reply by ktsangop** [reply]
Not sure I have the knowledge to do so but maybe I'll give it a try ..!

**Reply by ScriptedAlchemy** [reply]
Start a PR on the docs. I'm happy to review and add notes. Just stretched thin so if anyone can help it'll take pressure off me.

**Reply by ktsangop** [reply]
No worries, thank you again for your hard work on this project!
Module federation is truly a revelation for micro-frontend architecture...

Not a promise, but I'd love to try and contribute with a couple of paragraphs ;)
